client/softlayer_client.go

```diff
@@ -16,39 +16,42 @@ import (
 	softlayer "github.com/maximilien/softlayer-go/softlayer"
 )
 
+var DefaultHTTPClient = http.DefaultClient
```

Comment:
Why create an additional `NewSoftLayerClient` where the `HttpClient` is injected and this one by default uses the `http.DefaultClient`?

Since this one `NewSoftLayerClient(username, apiKey string) *SoftLayerClient` would still exist, current client satisfied with default would be OK. That way it's the responsibility of the caller to either create the HTTP client and pass it or use the default?

If you are OK with that please add some tests for both `NewSoftLayerClient` functions and rebase and I will merge.

Sorry about back and forth, want to make sure we have something we all like.

Thanks much. Best.